core/src/main/java/com/novoda/noplayer/internal/exoplayer/mediasource/MediaSourceFactory.java

```diff
@@ -44,16 +49,30 @@ public MediaSource create(Options options,
                               MediaSourceEventListener mediaSourceEventListener,
                               DefaultBandwidthMeter bandwidthMeter) {
         DefaultDataSourceFactory defaultDataSourceFactory = createDataSourceFactory(bandwidthMeter);
+
+        MediaSource contentMediaSource;
         switch (options.contentType()) {
             case HLS:
-                return createHlsMediaSource(defaultDataSourceFactory, uri, mediaSourceEventListener);
+                contentMediaSource = createHlsMediaSource(defaultDataSourceFactory, uri);
+                break;
             case H264:
-                return createH264MediaSource(defaultDataSourceFactory, uri, mediaSourceEventListener);
+                contentMediaSource = createH264MediaSource(defaultDataSourceFactory, uri);
+                break;
             case DASH:
-                return createDashMediaSource(defaultDataSourceFactory, uri, mediaSourceEventListener);
+                contentMediaSource = createDashMediaSource(defaultDataSourceFactory, uri);
+                break;
             default:
                 throw new UnsupportedOperationException("Content type: " + options + " is not supported.");
         }
+
+        if (advertsLoader.isPresent()) {
```

Comment:
If we are provided with an AdsLoader then we wrap the mediaSource with an AdsMediaSource to allow for ads playback